xarray/core/dataset.py

```diff
@@ -1491,7 +1491,15 @@ def __setitem__(self, key: Hashable, value) -> None:
                 "cannot yet use a dictionary as a key to set Dataset values"
             )
 
-        self.update({key: value})
+        if isinstance(key, list):
+            if len(key) != len(value):
+                raise ValueError("keys and value must have the same length.")
```

Comment:
Could we be slightly more descriptive — maybe give the length of each, if not the full values? I'm not sure `key` and `value` alone will be that clear b/c they're not named arguments.

Reply:
I've tried to be more descriptive and to give a more user-friendly message, although it isn't easy to be concise